src/components/Controls.svelte

```diff
@@ -27,38 +28,50 @@
 		if (currStatus == status.PLAY) currStatus = status.PAUSE;
 		else currStatus = status.PLAY;
 	}
-
-	//on currStatus value change
-	// if(currStatus == status.PLAY){
-	// 	audio.load()  might not need
-	//  audio.currentTime = trackPosition
-	// 	audio.play()
-	// }
-	// else{
-	// 	audio.pause()
-	//  trackPosition = audio.currentTime
-	// }
-
+	
 	function handleAddSong() {
-		console.log('added');
+		added = !added;
 	}
 	function handleLikeSong() {
-		console.log('liked');
+		liked = !liked;
+	}
+	function togglePlay(){
+		if(currStatus === status.PLAY){
+			currStatus = status.PAUSE;
+			audio.pause();
+			position = Math.round(audio.currentTime);
+		}
+		else{
+			currStatus = status.PLAY;
+			audio.currentTime = position;
+			audio.play();
+			setInterval(() => {
+				position = Math.round(audio.currentTime);
+				if(position >= audio.duration){
+					currStatus = status.PAUSE;
+					position = 0;
+				}
+			}, 500);
+		}
+	}
+	function seek(event:any){
+		audio.currentTime = event.target.value;
 	}
 </script>
 
-<div class="controls">
-	<div>
-		<button
-			><span class="material-icons"
-				>{currStatus === status.PLAY ? 'play_arrow' : 'pause'}</span
-			></button
-		>
-	</div>
-	<div>
-		<button on:keyup={handleLikeSong}
-			><span class="material-icons">thumb_up</span></button
-		>
-		<button on:keyup={handleAddSong}>add to playlist</button>
+<div class= "controls">
+	<div class="controlButtons">
+		<button class="playPause" on:click|stopPropagation={togglePlay}>
+			<span class="material-icons">
+				{currStatus === status.PLAY ? 'pause' : 'play_arrow'}
+			</span>
+		</button>
+		<input type="range" max=30 value={position} class="trackPlaceholder" on:click|stopPropagation|self={seek}/>
```

Comment:
hmmm this is similar to the `Slider` component that I am working on. I will make it more adaptable so it can also be used here.